From 704fd9e38ef0813e17bd2b662c41fecfa79f47c4 Mon Sep 17 00:00:00 2001
From: Carlo Dri <carlo.dri@gmail.com>
Date: Wed, 20 Aug 2025 22:08:14 +0200
Subject: [PATCH] replace wget with requests

---
 caerus/caerus.py | 10 +++++++---
 requirements.txt |  2 +-
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/caerus/caerus.py b/caerus/caerus.py
index 43c580d..9a51341 100644
--- a/caerus/caerus.py
+++ b/caerus/caerus.py
@@ -16,9 +16,9 @@
 # Custom helpers
 from caerus.utils.ones2idx import region2idx
 from caerus.utils.risk_performance_metrics import risk_performance_metrics
-import wget
 import os
 import matplotlib.pyplot as plt
+import requests
 pd.options.mode.chained_assignment = None  # default='warn'
 warnings.filterwarnings('ignore')
 
@@ -279,7 +279,8 @@ def download_example(self, name='btc', verbose=3):
             getfeat='Close'
 
         curpath = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
-        PATH_TO_DATA = os.path.join(curpath, wget.filename_from_url(url))
+        filename = os.path.basename(url)
+        PATH_TO_DATA = os.path.join(curpath, filename)
 
         # Create dir
         if not os.path.isdir(curpath):
@@ -288,7 +289,10 @@ def download_example(self, name='btc', verbose=3):
         # Check file exists.
         if not os.path.isfile(PATH_TO_DATA):
             if verbose>=3: print('[caerus] Downloading example dataset..')
-            wget.download(url, curpath)
+            response = requests.get(url)
+            response.raise_for_status()
+            with open(PATH_TO_DATA, 'wb') as f:
+                f.write(response.content)
 
         # Import local dataset
         if verbose>=3: print('[caerus] Import dataset..')